main.go

```diff
@@ -20,31 +27,32 @@ func main() {
 		panic("GROUP_NAME must be specified")
 	}
 
-	// This will register our custom DNS provider with the webhook serving
+	// This will register our NS1 DNS provider with the webhook serving
 	// library, making it available as an API under the provided GroupName.
 	// You can register multiple DNS provider implementations with a single
 	// webhook, where the Name() method will be used to disambiguate between
 	// the different implementations.
 	cmd.RunWebhookServer(GroupName,
-		&customDNSProviderSolver{},
+		&ns1DNSProviderSolver{},
 	)
 }
 
-// customDNSProviderSolver implements the provider-specific logic needed to
-// 'present' an ACME challenge TXT record for your own DNS provider.
-// To do so, it must implement the `github.com/jetstack/cert-manager/pkg/acme/webhook.Solver`
-// interface.
-type customDNSProviderSolver struct {
+// ns1DNSProviderSolver implements the logic needed to 'present' an ACME
+// challenge TXT record. To do so, it implements the
+// `github.com/jetstack/cert-manager/pkg/acme/webhook.Solver` interface.
+type ns1DNSProviderSolver struct {
```

Comment:
Dumb stylistic thing, but isn't the `DNSProvider` part of this name redundant?  Would `ns1Solver` make more sense?

Reply:
Eh, I'm seeing in their repo that the name `DNSProvider` probably has some sort of significance for that project.  Maybe this is a dumb suggestion in light of that.

Reply:
yeah. i want to minimize changes from the "template" as much as possible too, to make upstream merges, or comparing to other implementations, as easy as possible.